momepy/preprocessing.py

```diff
@@ -740,3 +746,313 @@ def _get_extrapolated_line(coords, tolerance, point=False):
     if point:
         return b
     return pygeos.linestrings([a, b])
+
+
+def _polygonize_ifnone(edges, polys):
+    if polys is None:
+        pre_polys = polygonize(edges.geometry)
+        polys = gpd.GeoDataFrame(geometry=[g for g in pre_polys], crs=edges.crs)
+    return polys
+
+
+def _selecting_rabs_from_poly(
+    gdf, circom_threshold=0.7, area_threshold=0.85, include_adjacent=True
+):
+    """
+    From a GeoDataFrame of polygons, returns a GDF of polygons that are
+    above the Circular Compaactness threshold.
+
+    Return
+    ________
+    GeoDataFrames : round abouts and adjacent polygons
+    """
+    # calculate parameters
+    gdf = gdf.copy()
+    gdf["area"] = gdf.geometry.area
+    gdf["circom"] = CircularCompactness(gdf, "area").series
```

Comment:
Strictly speaking, you don't need to assign these as columns because those boolean filters you are creating below will work as well with the series' directly. Then you can probably avoid the copy above.